lib/DDGC/Schema/Result/User/Subscription.pm

```diff
@@ -0,0 +1,89 @@
+package DDGC::Schema::Result::User::Subscription;
+# ABSTRACT: Activity feed subscription result class
+
+use Moo;
+extends 'DDGC::Schema::Result';
+use DBIx::Class::Candy;
+
+table 'user_subscription';
+
+column id => {
+    data_type => 'bigint',
+    is_auto_increment => 1,
+};
+primary_key 'id';
+
+column users_id => {
+    data_type   => 'bigint',
+    is_nullable => 0,
+};
+
+column category => {
+    data_type => 'text',
+    is_nullable => 0,
+};
+
+column action => {
+    data_type => 'text',
+    is_nullable => 1,
+};
+
+column meta1 => {
+    data_type => 'text',
+    is_nullable => 1,
+};
+
+column meta2 => {
+    data_type => 'text',
+    is_nullable => 1,
+};
+
+column meta3 => {
+    data_type => 'text',
+    is_nullable => 1,
+};
+
+belongs_to 'user', 'DDGC::Schema::Result::User', 'users_id';
+
+has_many 'activity', 'DDGC::Schema::Result::ActivityFeed', sub {
+    my ( $args ) = @_;
+
+    my $activity = $args->{foreign_alias};
+    my $subscription = $args->{self_alias};
+
+    +{
+        -and => {
+            "$activity.category" => "$subscription.category",
+            -or => [
+                "$subscription.action" => { '=' => undef },
+                "$activity.action" => "$subscription.action",
+            ],
+            -or => [
+                "$subscription.meta1" => { '=' => undef },
+                "$activity.meta1" => { 'like' => "$subscription.meta1" },
+            ],
+            -or => [
+                "$subscription.meta2" => { '=' => undef },
+                "$activity.meta2" => { 'like' => "$subscription.meta2" },
+            ],
+            -or => [
+                "$subscription.meta3" => { '=' => undef },
+                "$activity.meta3" => { 'like' => "$subscription.meta3" },
+            ],
+            -or => [
+                "$activity.for_user" => { '=' => undef },
+                "$activity.for_user" => { '=' => "$subscription.users_id" },
+            ],
+            -or => [
+                "$activity.for_role" => { '=' => undef },
+                "$activity.for_role" => {
+                    -in => \"SELECT role FROM user_role WHERE users_id = $subscription.users_id",
+                }
+            ]
+        }
+    };
+
+};
```

Comment:
Time to teach computers to feel pain, my [SQL::Abstract](https://metacpan.org/pod/SQL::Abstract)-fu failing me miserably here:

```sql
SELECT "me"."id", "me"."created", "me"."category", "me"."action", "me"."meta1", "me"."meta2", "me"."meta3", "me"."description", "me"."format", "me"."for_user", "me"."for_role" 
  FROM "user_subscription" "user_subscription__row" 
  JOIN "activity_feed" "me" 
    ON ( "me"."for_role" IS NULL OR "me"."for_role" IN ( 
        SELECT role 
          FROM user_role 
        WHERE users_id = user_subscription__row.users_id
       ) ) AND "me"."category" = 'user_subscription__row.category' 
WHERE "user_subscription__row"."id" = '5'
```